### DISCUSSION TITLE
Login and Authenticated

### DISCUSSION THREAD

**Discussion by genfish** [pimcore/pimcore#16526] [Q&A / Support]
Hi my friends, sorry for my english.

I maked some apps and webs baseds in Pimcore 2 to 5, and now I trying manually update these projectes to Pimcore 11.

The first problem I encounter is the new way to Login and Authenticated with Symfony. I don't understand how this Framework manage it. I can't find the files this process in workflow Symfony.

For example:

Before, I can do something like this:

```
Pimcore::getEventManager()->trigger("admin.login.login.authenticate", $this, [
      "username" => $username
,     "password" => $password
]);

$user = Pimcore_Tool_Authentication::authenticatePlaintext($username, $password);

if ( $user instanceof User) {
      Pimcore_Tool_Session::useSession(function($adminSession) use ($user) {
            $adminSession->user = $user;
            Pimcore_Tool_Session::regenerateId();
      });
}
```

Similar to how Pimcore handled the backend Login. But now all that doesn't exist anymore and the password doesn't go anywhere...

I understand it is now that using configurations in security.yaml and others?

I have seen several files and projectes... the "demo" from Pimcore and bundles admin-ui-classic-bundle, customer-data-framework, CoreShop, pimcore-members...

I haven't been able to determine how a simple Login to the Pimcore backend works so I can replicate it. Can anyone give me a clue or help?

**Comment by kg303** [comment]
There's not so much about login and authentication in Pimcore.

Look into this: https://pimcore.com/docs/platform/Pimcore/Development_Tools_and_Details/Security_Authentication/Authenticate_Pimcore_Objects/

Also, in Pimcore Demo you can find necessary files, such as App/Model/DataObject/User.php class, App/Form/LoginFormType and under App/Controller/AccountController.php you can take loginAction part and use it for yourself. 

And create CustomAuthenticator.php class under App/Security.

You will probably need to install symfony/form bundle in Pimcore since it is not included in Skeleton version of Pimcore. 
Symfony/form is used for your LoginForm and basically you can copy login.html.twig from Pimcore demo. 

And in the end, or at the beginning, you also need to configure your config files: config.yaml, services.yaml and security.yaml

**Reply by kg303** [reply]
And you could probably do it with CustomerManagementFramework bundle, creating Customer class in Pimcore and copying all files from Pimcore Demo.

**Comment by genfish** [comment]
Thanks for your reply, but it's not what I'm looking for.
Obviously if I install some versions of Pimcore like the Demo, Ecommerce or Customer, I have the Login configured, but it also fills the projects with many functionalities that I don't need and I still don't know how it really works.
I would like to find is a way to create it personalized and understand how the Login and Authentication process works.
I don't understand the need to create a User Object in Pimcore with username, password and roles (Authenticate_Pimcore_Objects), instead of using the platform's users and roles system directly and just binding it to an object to have custom data.
